server/src/main/java/io/crate/execution/engine/aggregation/impl/average/FractionalAverageStateType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.engine.aggregation.impl.average;
+
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+
+public class FractionalAverageStateType extends AverageStateType {
+
+    static final int ID = 1026;
```

Comment:
I'm afraid this will still break rolling upgrade compatibility. Old nodes won't be able to handle this new version id.

We'd to ensure that only the previous state type is used if one of the nodes is older than the current version.

Reply:
Alternative solution would be to parameterize the add function and keep a single state - given that the internal representation is the same in both cases. 

The current diff may be too influenced by the planned upcoming change to address the overflow handling.

Reply:
I went with version check approach - was indeed influenced by keeping overflow issue in mind. I swapped id-s of integral and fractional types - now fractional is 1024 (like old averageStateType) and it's BWC with it. And integralType now has 1026 and will be used for with OverflowAwareMutableLong later - it's compatible now but it definitely won't be compatible later

Reply:
I will revert part of my changes related to future overflow handling and get back to single average state to keep BWC, that would simplify my current PR and to handle precision I will add special numericavgagregation like we did for sum (with hint in docs) - it wouldn't break BWC as new argument -> return type is an extension.